src/dlgRoomProperties.cpp

```diff
@@ -0,0 +1,555 @@
+    }
+    return newSymbolText;
+}
+
+
+int dlgRoomProperties::getNewWeight()
+{
+    if (mpWeights.size() <= 1) {
+        return spinBox_weight->value();
+    }
+    QString newWeightText = comboBox_weight->currentText();
+    if (newWeightText == multipleValuesPlaceholder) {
+        return -1; // User did not want to select any weight, so we will do no change
+    }
+    // Parse an initial number out of what was selected or typed
+    QRegularExpression countStripper(qsl("^\\s*(\\d+)"));
+    QRegularExpressionMatch match = countStripper.match(newWeightText);
+    if (match.hasMatch() && match.lastCapturedIndex() > 0) {
+        return match.captured(1).toInt();
+    }
+    if (newWeightText.toInt() > 0) {
+        return newWeightText.toInt();
+    }
+    return -2; // Maybe some other input we did not understand, so we will do no change
+}
+
+
+void dlgRoomProperties::slot_updatePreview()
+{
+    auto realSymbolColor = selectedSymbolColor != nullptr ? selectedSymbolColor : defaultSymbolColor();
+    QString newSymbol = getNewSymbol();
+    if (newSymbol == multipleValuesPlaceholder) {
+        newSymbol = QString();
+        pushButton_setSymbolColor->setStyleSheet(QString());
+    } else {
+        pushButton_setSymbolColor->setStyleSheet(
+        qsl("background-color: %1; color: %2; border: 1px solid; border-radius: 1px;")
+            .arg(realSymbolColor.name(), backgroundBasedColor(realSymbolColor).name()));
+    }
+    label_preview->setFont(getFontForPreview(newSymbol));
+    label_preview->setText(newSymbol);
+    label_preview->setStyleSheet(
+        qsl("color: %1; background-color: %2; border: %3;")
+            .arg(realSymbolColor.name(), mRoomColor.name(), mpHost->mMapperShowRoomBorders ? qsl("1px solid %1").arg(mpHost->mRoomBorderColor.name()) : qsl("none")));
+
+}
+
+
+QFont dlgRoomProperties::getFontForPreview(QString symbolString)
+{
+    auto font = mpHost->mpMap->mMapSymbolFont;
+    font.setPointSize(font.pointSize() * 0.9);
+    if (!symbolString.isEmpty()) {
+        QFontMetrics mapSymbolFontMetrics = QFontMetrics(font);
+        QVector<quint32> codePoints = symbolString.toUcs4();
+        QVector<bool> isUsable;
+        for (int i = 0; i < codePoints.size(); ++i) {
+            isUsable.append(mapSymbolFontMetrics.inFontUcs4(codePoints.at(i)));
+        }
+        bool needToFallback = isUsable.contains(false);
+        if (needToFallback) {
+            symbolString = QString(QChar::ReplacementCharacter);
+            font.setStyleStrategy(static_cast<QFont::StyleStrategy>(mpHost->mpMap->mMapSymbolFont.styleStrategy() & ~(QFont::NoFontMerging)));
+        }
+    }
+    return font;
+}
+
+
+void dlgRoomProperties::slot_openSymbolColorSelector()
+{
+    auto* dialog = (selectedSymbolColor != nullptr) ? new QColorDialog(selectedSymbolColor, this) : new QColorDialog(defaultSymbolColor(), this);
+    dialog->setAttribute(Qt::WA_DeleteOnClose);
+    dialog->setWindowTitle(tr("Pick color"));
```

Comment:
Given that there is now two cases where a colour is to be selected in this dialogue would it be worth emphasising what the colour is being selected for:
```suggestion
    dialog->setWindowTitle(tr("Pick color for symbol"));
```

Reply:
Will review all dialog titles.